galter_subjects_utils/updater.py

```diff
@@ -133,17 +157,26 @@ def update_rdm_record(record, ops_data, logger, keep_trace):
                 keep_trace.trace(record, op_data["subject"])
             logger.log(record.pid.pid_value, delta=op_data)
 
+    # Make sure subjects are deduplicated
+    record["metadata"]["subjects"] = deduplicate_subjects(
+        record["metadata"]["subjects"]
+    )
+
     # All side-effects
+    # ---
     records_service = current_service_registry.get("records")
+    # According to other code in InvenioRDM, the same indexer is used for
+    # records and drafts
     commit_op = RecordCommitOp(record, records_service.indexer)
-    # the following on_register, on_commit don't use the uow object
+    # The following on_register, on_commit don't use the uow object
     # so passing None is fine
     fake_uow = None
     try:
         commit_op.on_register(fake_uow)  # commits to DB
         commit_op.on_commit(fake_uow)  # reindexes in index
     except Exception as e:
-        logger.log(record.pid.pid_value, error=str(e))
+        msg = re.sub(r"\s+", " ", str(e))
```

Comment:
Makes sure that the error message fits in one line and thus serializes to csv nicely.

Reply:
What alterted me to this was how the duplicate subjects error was written to the csv as the following:

```
9pm35-19s71,2024-03-05 08:37:25.341276,"[{'subject': 'COVID-19'}, {'id': 'https://id.nlm.nih.gov/mesh/D000086382'}, {'id': 'https://id.nlm.nih.gov/mesh/D003964'}, {'id': 'https://id.nlm.nih.gov/mesh/D003964'}, {'subject': 'COVID-19 (Disease)--Complications'}, {'id': 'https://id.loc.gov/authorities/subjects/sh85037585'}, {'subject': 'COVID-19/complications'}, {'subject': 'Diaphragm/physiopathology'}, {'subject': 'Diaphragm/diagnostic imaging'}] has non-unique elements

Failed validating 'uniqueItems' in schema['properties']['metadata']['properties']['subjects']:
    {'items': {'$ref': '#/subject'}, 'type': 'array', 'uniqueItems': True}

On instance['metadata']['subjects']:
    [{'subject': 'COVID-19'},
     {'id': 'https://id.nlm.nih.gov/mesh/D000086382'},
     {'id': 'https://id.nlm.nih.gov/mesh/D003964'},
     {'id': 'https://id.nlm.nih.gov/mesh/D003964'},
     {'subject': 'COVID-19 (Disease)--Complications'},
     {'id': 'https://id.loc.gov/authorities/subjects/sh85037585'},
     {'subject': 'COVID-19/complications'},
     {'subject': 'Diaphragm/physiopathology'},
     {'subject': 'Diaphragm/diagnostic imaging'}]",https://id.nlm.nih.gov/mesh/D000086382Q000150 -> https://id.nlm.nih.gov/mesh/D000086382 + https://id.nlm.nih.gov/mesh/D003964Q000503 -> https://id.nlm.nih.gov/mesh/D003964 + https://id.nlm.nih.gov/mesh/D003964Q000000981 -> https://id.nlm.nih.gov/mesh/D003964
```

Now it condenses it all to a message without newlines that fits correctly into the csv.

Reply:
> without newlines
ah, got it. I was thinking of the regex as identifying multiple white spaces rather than including new lines.